components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -534,7 +534,7 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
             if (nonce == null) {
                 LOG.debug("OIDC provider does not support nonce claim in id_token.");
             }
-            if (nonce != null && !nonce.equals(context.getProperty(NONCE))) {
+            if (nonce != null && context.getProperty(NONCE) != null && !nonce.equals(context.getProperty(NONCE))) {
```

Comment:
Lets add the null check before the `if (nonce == null) {` line. So the complete nonce validation portion will only engage if the nonce is available in the context

Reply:
Fixed by - https://github.com/wso2-extensions/identity-outbound-auth-oidc/pull/142/commits/014c26a150f1306923f029c49350e98e7c039f18